### DISCUSSION TITLE
Ignored Test cases && Gradle Test Executor error

### DISCUSSION THREAD

**Discussion by VermaSankalp** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#19] [Q&A]
When I run the 'Test All' run config, the GameBoardControllerTest test cases ignores 10 of the test cases (all the test cases apart from the sanity tests are ignored) and the 'Gradle Test Executor' also shows an error.

Anyone know why this might be happening? 

The GameBoardControllerTest test cases were not being ignored before but they are being ignored for some reason now.

**Comment by Derppening** [comment]
If there are test cases which are ignored, it is likely that there are parts of your implementation which are incorrect.

> [...] 'Gradle Test Executor' also shows an error

You might want to show what error you are getting.

**Reply by VermaSankalp** [reply]
<img width="1430" alt="Screenshot 2021-09-27 at 10 10 09 PM" src="https://user-images.githubusercontent.com/73076775/134925262-cb572b78-f651-400f-ac1c-cc9f68a4b131.png">

So the error is called 'Class Gradle Test Executor 31' and its content is like the screenshot but a lot more of them. In case the screenshot is not visible, this is the link to it: https://user-images.githubusercontent.com/73076775/134925262-cb572b78-f651-400f-ac1c-cc9f68a4b131.png.

**Reply by Derppening** [reply]
Reload/Reimport the Gradle project, run the `Clean` task, and try to run the test cases again. One of your fellow classmates also had the same issue and that is how I fixed it.

**Reply by VermaSankalp** [reply]
Could I reload the gradle project by clicking 'Reload from Disk' after right clicking the gradle folder in the main directory?

Edit: Nevermind, found out how to do it. It seems to have worked the first time when I run 'Test all' after I reloaded the gradle project and ran 'clean', but when i ran 'Test all' a second time, it went back to ignoring the test cases.

Do I need to reload and run clean everytime I want the test cases to all work?

**Reply by VermaSankalp** [reply]
![Screenshot 2021-09-27 at 10 36 25 PM](https://user-images.githubusercontent.com/73076775/134929964-feabb130-5cd4-42bc-9028-545a6f2775e6.png)

Also when I try to run with Junit, there is the message that shows up with the blocked test cases. In the case of the screenshot, line 220 in GameBoardControllerTest.java is: 
assumeTrue(gameBoard.getPlayer().equals(gameBoard.getEntityCell(0, 0).getEntity()));

Does this have anything to do with my implementation or is it some external issue?

**Reply by Derppening** [reply]
The assumeTrue statement is something I have put in the test case to make sure that if your implementation is faulty, I do not deduct your marks twice.

For example, if you have a faulty implementation of GameBoard, in theory both Gameboard and GamboardController tests would fail, causing you to lose a lot of marks.

By using assumptions, I am able to instead identify that your GameBoard implementation is incorrect, and perform some additional checks to perhaps recover some points for you.

TL;DR If you get some test(s) ignored due to `assumeTrue`/`assumeFalse`, your implementation is incorrect.

**Reply by Derppening** [reply]
Also, if "Test All" is nondeterministic like you mentioned, there is a chance that your project might not be configured correctly. You may want to arrange a meeting with me to deal with this issue.

**Reply by VermaSankalp** [reply]
I identified the source of the blocked cases and fixed it, and after reloading gradle and running clean, my test cases are 100%. Thanks!
